features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Overview.jsx

```diff
@@ -165,6 +165,89 @@ function Overview() {
         return filteredApiPolicies && filteredApiPolicies.length > 0;
     };
 
+    const updateSelectedEndpoint = (e) => {
+        const selectedEnvName = e.target.value;
+        const filteredEndpoints = api.endpointURLs.filter((ep) => ep.environmentName === selectedEnvName);
+        if (filteredEndpoints && filteredEndpoints.length > 0) {
+            setSelectedEndpoint(filteredEndpoints[0]);
+        } else {
+            Alert.error(intl.formatMessage({
+                id: 'Apis.Details.Overview.select.env.error',
+                defaultMessage: 'Error Selecting Environment',
+            }));
+        }
+    };
+
+    const getSubscriptionPolicies = async () => {
+        const restApi = new API();
+        return restApi.getAllTiers('subscription')
+            .then((response) => {
+                try {
+                // Filter policies base on async or not.
+                    const filteredList = response.body.list.filter((str) => isApiPolicy(str.name));
+                    setAllPolicies(filteredList);
+                } catch (e) {
+                    console.log(e);
+                    Alert.error(intl.formatMessage({
+                        id: 'Apis.Details.Overview.error.occurred',
+                        defaultMessage: 'Error occurred',
+                    }));
+                }
+            }).catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
+                }
+                const { status } = error;
+                if (status === 404) {
+                    Alert.error(intl.formatMessage({
+                        id: 'Apis.Details.Overview.error.occurred',
+                        defaultMessage: 'Error occurred',
+                    }));
+                    setNotFound(true);
+                }
+                setAllDocuments([]);
+            });
+    };
+
+    const getDocuments = async () => {
+        const restApi = new API();
+        return restApi.getDocumentsByAPIId(api.id)
```

Comment:
You have defined an `async` method but using promise call backs in `then` & `catch` Shall we stick to one method either `async/await` or `Promise`

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/10298/commits/fd68d8dfc8f1759263ac96dd2593dbd1842b1bdb